patroni/ha.py

```diff
@@ -584,7 +584,9 @@ def _get_node_to_follow(self, cluster: Cluster) -> Union[Leader, Member, None]:
         """
         # The standby leader or when there is no standby leader we want to follow
         # the remote member, except when there is no standby leader in pause.
-        if self.is_standby_cluster() and (self.has_lock(False) or self.cluster.is_unlocked() and not self.is_paused()):
+        if self.is_standby_cluster() \
+                and (cluster.leader and cluster.leader.name and cluster.leader.name == self.state_handler.name
```

Comment:
Btw, why do we use `cluster.leader` here but `self.cluster.is_unlocked()` immediately after that ?

Reply:
Addressed in https://github.com/zalando/patroni/pull/2900/commits/47e590dc153f0078203d9da57e6df8894575e731